modules/colors.html

```diff
@@ -0,0 +1,212 @@
+          );
+        background-size: 100% 100%, 20px 20px;
+        padding: 0;
+        margin: 0;
+      }
+      div {
+        width: 100vw;
+        height: 100vh;
+      }
+    </style>
+  </head>
+  <body>
+    <div>
+      <dialog>
+        <table>
+          <caption>
+            Current color values:
+          </caption>
+          <tbody>
+            <tr id="RGB">
+              <th>RGB</th>
+              <td></td>
+            </tr>
+            <tr id="HEX">
+              <th>HEX</th>
+              <td></td>
+            </tr>
+            <tr id="HSL">
+              <th>HSL</th>
+              <td></td>
+            </tr>
+            <tr id="HWB">
+              <th>HWB</th>
+              <td></td>
+            </tr>
+          </tbody>
+        </table>
+
+        <p>
+          <label for="color">Select a color: </label
+          ><input type="color" id="color" value="#bada55" />
+        </p>
+        <p>
+          <label for="opacity">Select an opacity: </label
+          ><input
+            type="range"
+            id="opacity"
+            value="1"
+            min="0"
+            max="1"
+            step="0.1"
+          />
+        </p>
+      </dialog>
+    </div>
+    <script>
+      const colorPicker = document.getElementById("color");
+      const opacityPicker = document.getElementById("opacity");
+
+      document.addEventListener("readystatechange", () => {
+        document.querySelector("dialog").show();
+        createColor();
+      });
+
+      colorPicker.addEventListener("change", () => {
+        createColor();
+      });
+
+      opacityPicker.addEventListener("change", () => {
+        createColor();
+      });
+
+      function createColor() {
+        const currentColor = colorPicker.value;
+        const currentOpacity = opacityPicker.value * 1;
+        setBackgroundColor(currentColor, currentOpacity);
+        createHEX(currentColor, currentOpacity);
+        createRGB(currentColor, currentOpacity);
+      }
+
+      function createHEX(color, opacity) {
+        const cell = document.querySelector("#HEX td");
+        if (opacity === 1) {
+          cell.textContent = color;
+        } else {
+          cell.textContent = hexOpacity(color, opacity);
+        }
+      }
+
+      function createRGB(color, opacity) {
+        const cell = document.querySelector("#RGB td");
+        color = color.substring(1, 7);
+        const hexArray = color.match(/.{1,2}/g);
+        const R = parseInt(hexArray[0], 16);
+        const G = parseInt(hexArray[1], 16);
+        const B = parseInt(hexArray[2], 16);
+
+        if (opacity === 1) {
+          cell.textContent = `rgb(${R} ${G} ${B})`;
+        } else {
+          cell.textContent = `rgb(${R} ${G} ${B} / ${opacity})`;
+        }
+        createHSL(R, G, B, opacity);
+      }
+
+      function createHSL(r, g, b, opacity) {
+        const cell = document.querySelector("#HSL td");
+        (r = r / 255), (g = g / 255), (b = b / 255);
```

Comment:
This is difficult to read.
```suggestion
        // Let's have r, g, b in the range [0, 1]
        r = r / 255;
        g = g / 255;
        b = b / 255;
```